src/main/clojure/aerospike_clj/client.clj

```diff
@@ -172,390 +105,367 @@
+
+  (operate [this index set-name expiration operations]
+    (pt/operate this index set-name expiration operations {}))
+
+  (operate [this index set-name expiration operations conf]
+    (if (empty? operations)
+      (p/resolved nil)
+      (let [op-future  (p/deferred)
+            start-time (System/nanoTime)]
+        (.operate ^AerospikeClient client
+                  ^EventLoop (.next ^EventLoops el)
+                  (AsyncRecordListener. op-future)
+                  ^WritePolicy (:policy conf (policy/write-policy client expiration RecordExistsAction/UPDATE))
+                  ^Key (pt/create-key index (:dbns this) set-name)
+                  (utils/v->array Operation operations))
+        (register-events (p/then op-future record/record->map) client-events "operate" index start-time))))
+
+  (scan-set [_this aero-namespace set-name conf]
+    (when-not (fn? (:callback conf))
+      (throw (IllegalArgumentException. "(:callback conf) must be a function")))
+    (let [op-future  (p/deferred)
+          start-time (System/nanoTime)
+          bin-names  (:bins conf)]
+      (.scanAll ^AerospikeClient client
+                ^EventLoop (.next ^EventLoops el)
+                (AsyncRecordSequenceListener. op-future (:callback conf))
+                ^Policy (:policy conf (ScanPolicy.))
+                aero-namespace
+                set-name
+                (when bin-names ^"[Ljava.lang.String;" (utils/v->array String bin-names)))
+      (register-events op-future client-events :scan nil start-time)))
+
+  pt/AerospikeAdminOps
+  (info [this node info-commands]
+    (pt/info this node info-commands {}))
+
+  (info [_this node info-commands conf]
+    (let [op-future  (p/deferred)
+          start-time (System/nanoTime)]
+      (.info ^AerospikeClient client
+             ^EventLoop (.next ^EventLoops el)
+             (AsyncInfoListener. op-future)
+             ^InfoPolicy (:policy conf (.infoPolicyDefault ^AerospikeClient client))
+             ^Node node
+             (into-array String info-commands))
+      (register-events op-future client-events :info nil start-time)))
+
+  (get-nodes [_this]
+    (into [] (.getNodes ^AerospikeClient client)))
+
+  (get-cluster-stats [_this]
+    (-> (.getClusterStats ^AerospikeClient client)
+        metrics/construct-cluster-metrics
+        metrics/cluster-metrics->dotted))
+
+  (healthy? [this]
+    (pt/healthy? this 1000))
+
+  (healthy? [this operation-timeout-ms]
+    (let [read-policy (let [p ^Policy (.readPolicyDefault ^AerospikeClient client)]
+                        (set! (.totalTimeout p) operation-timeout-ms)
+                        p)
+          k           (str "__health__" (rand-int Integer/MAX_VALUE))
+          v           (rand-int Integer/MAX_VALUE)
+          ttl         (min 1 (int (/ operation-timeout-ms 1000)))
+          set-name    "__health-check"]
+      (try
+        @(pt/put this k set-name v ttl)
```

Comment:
Why are we blocking here?

Reply:
I don't think there's much meaning to an asynchronous health check. When you ask the question "are you healthy?" you're gonna wait until you get an answer.